Fix Test Issue in LKSM

labkey-danield opened the pull request:
#### Rationale
There are some intermittent failure in the lineage tests in SampleManager. These failures happen because a parent sample/source was not removed when editing a child sample. I suspect that the issue is the ReactControl did not successfully remove the selected item from the control. This adds code in the ParentEntityEditPanel to check that an update to a parent, add or remove, happened. If the addition or removal is not successful an exception will be thrown. This will cause the test to fail at that time and not later on where it is more difficult to debug.

All of the Biologics, Labbook, Inventory and SampleManager tests have been run on TC using this change/branch.

#### Related Pull Requests
* TBD

#### Changes
* Added a waitFor check in the addParents method.
* Added a waitFor check in the removeParent method.

labkey-danield committed:
Merge branch 'develop' into fb_bootstrapMenuFix

labkey-danield committed:
Add screenshots (on error) to LimitedRead tests.
Added check to make sure parents are added or removed from a sample's lineage.
Updaed a comment.

labkey-danield committed:
Increase wait time for parent to be removed.

labkey-danield committed:
Use an implied retry to delete a parent.

labkey-danield committed:
Improve check for parent removed.

labkey-danield committed:
Improve check that parent has been removed.

labkey-danield committed:
Merge branch 'develop' into fb_bootstrapMenuFix

labkey-danield committed:
Adding some comments.

labkey-danield committed:
Merge branch 'develop' into fb_bootstrapMenuFix

labkey-tchad commented on src/org/labkey/test/components/ui/entities/ParentEntityEditPanel.java:
```diff
@@ -423,7 +426,23 @@ public ParentEntityEditPanel removeEntityType(String typeName)
+                {
+                    // Guard against react selector being in a 'Loading...' state.
+                    WebDriverWrapper.waitFor(()->getParent(parentEntity).getComponentElement().isDisplayed(),
+                            String.format("Selector for parents from '%s' is not visible.", parentEntity), 1_000);
```
This shouldn't be necessary. `BaseReactSelect.removeSelection` already does this wait (plus more).
```suggestion
```

labkey-danield replied:
I've removed my changes to the ParentEntityEditPanel, and now only have a check in the BaseReactSelect to fail if it does not remove a selected element in the multi select control.
This should help debug random/intermittent failures in LKSM where lineage updates are not as expected.

labkey-tchad reviewed (commented):
Discussed moving change to `BaseReactSelect`. Ping for review once that's ready.

labkey-danield committed:
Reconsidering change to ParentEntityPanel. Adding a check to baseReactSelect.removeSelection to validate that selection was removed.
Commenting out previous change for the time being.

labkey-danield committed:
Removed changes that were commented out.
Make check for BaseReactSelect.removeSelection to not check staleness of element, only check selection list.
Update check in FilteringReactSelect.typeAheadSelect that value was added. (Make sure other items were not removed).
Implemented some linter suggestions in FilteringReactSelect.

labkey-danield committed:
Rolling back changes made to FilteringReactSelect.

labkey-danield committed:
Revert all changes to ParentEntityEditPanel.

labkey-danield committed:
Trying to make delete a little more reliable.

labkey-danield committed:
Another attempt to make delete more reliable.

labkey-danield committed:
Merge branch 'develop' into fb_bootstrapMenuFix
